### PR TITLE
Add teachercon mapping to Pd teacher application

### PR DESCRIPTION
**Pull request by aoby**
Now, the teacher application form data will store the set of partner local workshop ids, or the teachercon details, from the time of application so we have that record.

### CODE REVIEW

**Comment by aoby** [comment]
[Teacher application: G3 workshop assignments](https://trello.com/c/0PfLcW8H/280-teacher-application-g3-workshop-assignments)

**Comment by aoby** [comment]
Good question. I suppose we could hardcode the id in the mapping data? Send me the ids and I'll add them. Note it will only make sense on production, but then I suppose that's the same for the partner names.

**Comment by Hamms** [comment]
We could also query for them by course and year, which might make implementing next year's version that much easier.

Either way, here they are
TeacherCon Atlanta: June 17-22
https://studio.code.org/pd/workshop_dashboard/workshops/3471
https://studio.code.org/pd/workshop_dashboard/workshops/3472

TeacherCon Phoenix: July 22-27
https://studio.code.org/pd/workshop_dashboard/workshops/3473
https://studio.code.org/pd/workshop_dashboard/workshops/3474

**Comment by aoby** [comment]
I added a `find_teachercon_workshop` method that takes course and city, and optionally year (default 2018).

File: dashboard/lib/pd/application/regional_partner_teachercon_mapping.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    def get_matching_teachercon(regional_partner)
+      return nil if regional_partner.nil?
+      REGIONAL_PARTNER_TC_MAPPING[regional_partner.name]
+    end
+
+    def find_teachercon_workshop(course:, city:, year: 2018)
+      Pd::Workshop.
+        # where(course: course, subject: Pd::Workshop::SUBJECT_TEACHER_CON).
+        where(course: course, subject: 'Code.org TeacherCon').
```
Comment (Hamms) [inline dashboard/lib/pd/application/regional_partner_teachercon_mapping.rb]: looks like some unfinished work here

Comment (aoby) [reply dashboard/lib/pd/application/regional_partner_teachercon_mapping.rb]: Oops. Thanks for catching. Will fix...

**Pull request merged**